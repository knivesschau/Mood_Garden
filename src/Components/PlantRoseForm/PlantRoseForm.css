/* mobile first style, min 320px */
.PlantRose {
    box-sizing: border-box;
}

.PlantRoseForm {
    text-align: center;
}

.Date_Entry, .Thorn_Entry, .Rose_Color, .Rose_Entry, .Bud_Entry {
    box-shadow: 0px 1px 5px 2px #6E7C51; 
    border-radius: 2em;
    margin: 1em;
    padding-top: .50em;
    background-color: rgb(203, 211, 167, .90);
}

#plant-title {
    text-align: center;
    margin-bottom: -.15em;
}

#date-entry-header, #plant-color-header, #plant-header-rose, #plant-header-bud, #plant-header-thorn {
    color: #6B1A1A;
    font-weight: bold;
    font-size: 1.5em;
}

#entry-date {
    font-size: 1em;
    width: 65%;
    padding: .30em;
    border-radius: 1em;
    margin-bottom: 1em;
    background-color: white;
    transition: border .3s ease-out; 
}

#date-entry-header {
    word-spacing: -.15em;
    margin-top: 0;
    margin-bottom: .30em;
}

#rose-prompt, #thorn-prompt, #bud-prompt, #color-prompt {
    font-size: 1em;
    margin-left: 1em;
    margin-right: 1em;
    margin-top: -1em;
    margin-bottom: .50em;
}

#colors {
    display: inline-block;
    font-size: 1em;
    width: 70%;
    padding: .30em;
    border-radius: 2em;
    background-color: white;
}

textarea {
    font-size: 1.10em;
    border-radius: .5em;
    height: 10em;
    width: 90%;
    margin-bottom: 1em;
    resize: none;
    transition: border .3s ease-out; 
}

textarea:focus, #colors:focus, input[type="date"]:focus {
    border: 10px solid #B00000;
    border-style: none none none solid;
    background-color: white;
}

#rose-icon, #thorn-icon, #bud-icon, #color-icon {
    margin-bottom: -.40em;
}

#submit-rose, #cancel-plant {
    display: inline-block;
    padding: .5em;
    margin-right: 1em;
    margin-bottom: .75em;
    border: 2px solid rgba(255,255,255,0);
    border-radius: 2em;
    box-sizing: border-box;
    margin-top: 1em;
    background-color: #C34444;
    color: #ffffff;
    font-weight: bold;
    font-size: 1em;
    text-shadow: 0 .65px 0.65px rgb(0,0,0,0.35);
}

#submit-rose:hover, #cancel-plant:hover {
    cursor: pointer;
    border-color: rgb(255,255,255, 1);
}

#submit-rose:active, #cancel-plant:hover {
    background-color: #8F0000;
}

@media only screen and (min-device-width: 375px) {
    #entry-date {
        width: 55%;
    }
}

@media only screen and (min-device-width: 414px) {
    .Date_Entry, .Thorn_Entry, .Rose_Color, .Rose_Entry, .Bud_Entry {
        margin: 1.5em;
    }
    
    #date-entry-header, #plant-color-header, #plant-header-rose, #plant-header-bud, #plant-header-thorn {
        font-size: 1.75em;
    }

    textarea, #entry-date, #rose-prompt, #thorn-prompt, #bud-prompt, #color-prompt, #submit-rose, #cancel-plant, #colors {
        font-size: 1.20em;
    }

    #plant-title {
        font-size: 3em;
    }
}

@media only screen and (min-device-width: 768px) {
    .Date_Entry, .Thorn_Entry, .Rose_Color, .Rose_Entry, .Bud_Entry {
        margin-left: 7.5em;
        margin-right: 7.5em;
    }

    #submit-rose, #cancel-plant {
        margin-left: -.50em;
        margin-right: 1.5em;
    }
}

@media only screen and (min-device-width: 1000px) {
    .Date_Entry, .Thorn_Entry, .Rose_Color, .Rose_Entry, .Bud_Entry {
        margin-left: 15em;
        margin-right: 15em;
    }
    
    #entry-date {
        width: 40%;
    }

    #submit-rose, #cancel-plant {
        margin-left: .10em;
    }

    #plant-title {
        font-size: 3.5em;
    }

    #date-entry-header, #plant-color-header, #plant-header-rose, #plant-header-bud, #plant-header-thorn {
        font-size: 2.5em;
    }

    textarea, #date-entry, #rose-prompt, #thorn-prompt, #bud-prompt, #color-prompt, #submit-rose, #cancel-plant, #colors {
        font-size: 1.5em;
    }
}

@media only screen and (min-device-width: 1200px) {
    .Date_Entry, .Thorn_Entry, .Rose_Color, .Rose_Entry, .Bud_Entry {
        margin-left: 20em;
        margin-right: 20em;
    }
}

@media only screen and (min-device-width: 1700px) {
    textarea {
        width: 95%;
    }
    
    .Date_Entry, .Thorn_Entry, .Rose_Color, .Rose_Entry, .Bud_Entry {
        margin-left: 30em;
        margin-right: 30em;
    }

    #plant-title {
        margin-bottom: .50em;
    }

    #entry-date {
        width: 35%;
    }
}

@media only screen and (min-device-width: 2000px) {
    .Date_Entry, .Thorn_Entry, .Rose_Color, .Rose_Entry, .Bud_Entry {
        margin-left: 35em;
        margin-right: 35em;
    }

    #entry-date {
        width: 30%;
    }

    #colors {
        width: 50%;
    }
}