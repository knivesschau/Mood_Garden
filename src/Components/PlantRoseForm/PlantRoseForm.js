import React, {Component} from 'react';
import moodGardenContext from '../../moodGardenContext';
import config from '../../config';
import './PlantRoseForm.css';

export default class PlantRoseForm extends Component {
    static defaultProps = {
        history: {
            push: () => {}
        },
    }

    static contextType = moodGardenContext;
    
    handleSubmit = e => {
        e.preventDefault();

        const newEntry = {
            entry_date: new Date(),
            rose: e.target['rose-text'].value,
            thorn: e.target['thorn-text'].value,
            bud: e.target['bud-text'].value,
            color: e.target['rose-colors'].value,
        };

        fetch(`${config.API_ENDPOINT}/roses`, {
            method: 'POST',
            headers: {
                'content-type': 'application/json'
            },
            body: JSON.stringify(newEntry)
        })
        .then(res => {
            if (!res.ok) {
                return res.json().then(e => Promise.reject(e))
            }
            return res.json();
        })
        .then(rose => {
            this.context.addEntry(rose);
            this.props.history.push(`/roses/${rose.id}`)
        })
        .catch(error => {
            console.error({error})
        });
    }
    
    render() {

        return (
            <section className="PlantRose">

                <h1>Plant a Rose</h1>

                <form className="PlantRoseForm" onSubmit={this.handleSubmit}>

                    <div className="Date_Entry">
                        <p id="date-entry">Date Planted:</p>

                        <input type="number" placeholder="01" min="01" max="12" required="" name="month" id="month"/>
                        <input type="number" placeholder="05" min="01" max="31" required="" name="day" id="day"/>
                        <input type="number" placeholder="2019" min="2019" max="2030" required="" name="year" id="year"/>

                    </div>

                    <div className="Rose_Entry">
                        <h2>Rose</h2>

                        <p id="rose-prompt">What is something you are looking forward to? Are there any new ideas that you would like to see grow?</p>
                        
                        <textarea name="rose-text" id="rose-entry-input" rows="15" placeholder="Write your thoughts here"/>
                    </div>

                    <div className="Thorn_Entry">
                        <h2>Thorn</h2>

                        <p id="thorn-prompt">What didn't go as planned? What stressed you out or made you feel upset?</p>

                        <textarea name="thorn-text" id="thorn-entry-input" rows="15" placeholder="Write your thoughts here"/>
                    </div>

                    <div className="Bud_Entry">
                        <h2>Bud</h2>

                        <p id="bud-prompt">What is something you are looking forward to? Are there any new ideas that you would like to see grow?</p>
                    
                        <textarea name="bud-text" id="bud-entry-input" rows="15" placeholder="Write your thoughts here"/>
                    </div>

                    <div className="Rose_Color">
                        <h2>Rose Color</h2>

                        <p id="color-prompt">Lastly, pick a rose color based on your current mood.</p>

                        <select name="rose-colors" id="colors">
                            <option value="Red">Red (Neutral)</option>
                            <option value="Yellow">Yellow (Happy)</option>
                            <option value="White">White (Thoughtful)</option> 
                            <option value="Pink">Pink (Grateful)</option>
                            <option value="Purple">Purple (Angry)</option>
                            <option value="Black">Black (Sad)</option>
                        </select> 

                    </div>
                    
                <button type="submit" id="submit-rose">Plant Rose</button>
                <button type="reset" id="start-over">Start Over</button>
                
                </form>

            </section>
        );
    }
}