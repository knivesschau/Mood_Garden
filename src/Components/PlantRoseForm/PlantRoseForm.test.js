import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import {BrowserRouter} from 'react-router-dom';
import toJson from 'enzyme-to-json';
import PlantRoseForm from './PlantRoseForm';

describe ('PlantRoseForm Component', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        
        ReactDOM.render(
            <BrowserRouter>
                <PlantRoseForm/>
            </BrowserRouter>, div)

        ReactDOM.unmountComponentAtNode(div)
    });

    it ('renders the complete PlantRoseForm Component', () => {
        const wrapper = shallow(<PlantRoseForm/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
})