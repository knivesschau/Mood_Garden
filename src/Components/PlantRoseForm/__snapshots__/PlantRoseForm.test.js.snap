// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantRoseForm Component renders the complete PlantRoseForm Component 1`] = `
<section
  className="PlantRose"
>
  <RoseNav />
  <h1
    id="plant-title"
  >
    Plant a Rose
  </h1>
  <form
    className="PlantRoseForm"
    onSubmit={[Function]}
  >
    <div
      className="Date_Entry"
    >
      <label
        htmlFor="date"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          color="green"
          fixedWidth={false}
          flip={null}
          icon="seedling"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="3x"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
        <h2
          id="date-entry-header"
        >
          Date Planted
        </h2>
        <input
          id="entry-date"
          name="date"
          required={true}
          type="date"
        />
      </label>
    </div>
    <div
      className="Rose_Entry"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        color="#FFFAEA"
        fixedWidth={false}
        flip={null}
        icon="grin-beam"
        id="rose-icon"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="3x"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <h2
        id="plant-header-rose"
      >
        Rose
      </h2>
      <label
        htmlFor="rose-entry-input"
      >
        <p
          id="rose-prompt"
        >
          What brought you happiness today? Did you make progress on any ideas or goals?
        </p>
        <textarea
          id="rose-entry-input"
          name="rose-text"
          onChange={[Function]}
          placeholder="Write your thoughts here"
          required={true}
          rows="15"
          value=""
        />
        <ErrorValidation
          valid={false}
        />
      </label>
    </div>
    <div
      className="Thorn_Entry"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        color="#A272AA"
        fixedWidth={false}
        flip={null}
        icon="frown"
        id="thorn-icon"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="3x"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <h2
        id="plant-header-thorn"
      >
        Thorn
      </h2>
      <label
        htmlFor="thorn-entry-input"
      >
        <p
          id="thorn-prompt"
        >
          Was there anything that upset or frustrated you today? What were the sources of those feelings?
        </p>
        <textarea
          id="thorn-entry-input"
          name="thorn-text"
          onChange={[Function]}
          placeholder="Write your thoughts here"
          required={true}
          rows="15"
          value=""
        />
        <ErrorValidation
          valid={false}
        />
      </label>
    </div>
    <div
      className="Bud_Entry"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        color="#8B5151"
        fixedWidth={false}
        flip={null}
        icon="grin-alt"
        id="bud-icon"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="3x"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <h2
        id="plant-header-bud"
      >
        Bud
      </h2>
      <label
        htmlFor="bud-text"
      >
        <p
          id="bud-prompt"
        >
          What is something you are looking forward to? What would you like to spend more time focusing on?
        </p>
        <textarea
          id="bud-entry-input"
          name="bud-text"
          onChange={[Function]}
          placeholder="Write your thoughts here"
          required={true}
          rows="15"
          value=""
        />
        <ErrorValidation
          valid={false}
        />
      </label>
    </div>
    <div
      className="Rose_Color"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        color="#B00000"
        fixedWidth={false}
        flip={null}
        icon="spa"
        id="color-icon"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="3x"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <h2
        id="plant-color-header"
      >
        Rose Color
      </h2>
      <label
        htmlFor="colors"
      >
        <p
          id="color-prompt"
        >
          Lastly, pick a rose color based on your current mood.
        </p>
        <select
          id="colors"
          name="rose-colors"
          onChange={[Function]}
          required={true}
          value=""
        >
          <option
            value=""
          />
          <option
            value="Red"
          >
            Red (Neutral)
          </option>
          <option
            value="Yellow"
          >
            Yellow (Happy)
          </option>
          <option
            value="White"
          >
            White (Thoughtful)
          </option>
          <option
            value="Pink"
          >
            Pink (Grateful)
          </option>
          <option
            value="Purple"
          >
            Purple (Angry)
          </option>
          <option
            value="Black"
          >
            Black (Sad)
          </option>
        </select>
        <ErrorValidation
          valid={false}
        />
      </label>
      <button
        id="submit-rose"
        type="submit"
      >
        Plant Rose
      </button>
      <button
        id="cancel-plant"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</section>
`;
